Bluetooth Crash

I connected my postmarketos device and was using it to play audio. Bluetooth just crashed after this though. Crash log:
```
type: crash
osVersion: google/panther/panther:15/BP1A.250405.007.B1/2025041100:user/release-keys
flags: dev options enabled
package: com.android.bluetooth:35, targetSdk 35
sharedUid: android.uid.bluetooth
process: com.android.bluetooth
processUptime: 26214 + 565 ms

android.os.DeadSystemRuntimeException: android.os.DeadSystemException
	at android.media.session.MediaController.getMetadata(MediaController.java:171)
	at com.android.bluetooth.audio_util.MediaController.getMetadata(MediaController.java:82)
	at com.android.bluetooth.audio_util.MediaPlayerWrapper.getMetadata(MediaPlayerWrapper.java:124)
	at com.android.bluetooth.audio_util.MediaPlayerWrapper.isMetadataSynced(MediaPlayerWrapper.java:270)
	at com.android.bluetooth.audio_util.MediaPlayerWrapper$MediaControllerListener.trySendMediaUpdate(MediaPlayerWrapper.java:442)
	at com.android.bluetooth.audio_util.MediaPlayerWrapper$MediaControllerListener.onPlaybackStateChanged(MediaPlayerWrapper.java:528)
	at android.media.session.MediaController$MessageHandler.handleMessage(MediaController.java:1202)
	at android.os.Handler.dispatchMessage(Handler.java:109)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.app.ActivityThread.main(ActivityThread.java:8973)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:591)
	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:369)
Caused by: android.os.DeadSystemException
	... 16 more
```

Logs:

[Bluetooth%20log%20b2b36fa1eb9e.txt](https://github.com/user-attachments/files/19721867/Bluetooth.20log.20b2b36fa1eb9e.txt)

--- a/media/java/android/media/MediaMetadata.java
+++ b/media/java/android/media/MediaMetadata.java
@@ -963,11 +963,15 @@ public final class MediaMetadata implements Parcelable {
             if (mBitmapDimensionLimit != Integer.MAX_VALUE) {
                 for (String key : mBundle.keySet()) {
                     Object value = mBundle.get(key);
-                    if (value instanceof Bitmap) {
-                        Bitmap bmp = (Bitmap) value;
+                    if (value instanceof Bitmap bmp) {
+                        final Bitmap orig = bmp;
                         if (bmp.getHeight() > mBitmapDimensionLimit
                                 || bmp.getWidth() > mBitmapDimensionLimit) {
-                            putBitmap(key, scaleBitmap(bmp, mBitmapDimensionLimit));
+                            bmp = scaleBitmap(bmp, mBitmapDimensionLimit);
+                        }
+                        Bitmap sharedBmp = bmp.asShared();
+                        if (orig != sharedBmp) {
+                            putBitmap(key, sharedBmp);
                         }
                     }
                 }

